main.go

```diff
@@ -560,10 +563,19 @@ func (c *controller) sync(ctx context.Context) {
 
 	c.populate(hashrings, statefulsets)
 
-	if err := c.saveHashring(ctx, hashrings, cm); err != nil {
+	err = c.saveHashring(ctx, hashrings, cm)
+	if err != nil {
 		c.reconcileErrors.WithLabelValues(save).Inc()
 		level.Error(c.logger).Log("msg", "failed to save hashrings")
 	}
+
+	// If enabled and hashring was succesfully changed, annotate pods with config hash on change.
```

Comment:
Lint fails due to this.
```suggestion
	// If enabled and hashring was successfully changed, annotate pods with config hash on change.
```